runtime/interpreter/interpreter.go

```diff
@@ -978,6 +980,51 @@ func (interpreter *Interpreter) declareAttachmentValue(
 	return interpreter.declareCompositeValue(declaration, lexicalScope)
 }
 
+// evaluateDefaultDestroyEvent evaluates all the implicit default arguments to the default destroy event.
+//
+// the handling of default arguments makes a number of assumptions to simplify the implementation;
+// namely that a) all default arguments are lazily evaluated at the site of the invocation,
+// b) that either all the parameters or none of the parameters of a function have default arguments,
+// and c) functions cannot currently be explicitly invoked if they have default arguments
+//
+// if we plan to generalize this further, we will need to relax those assumptions
+func (interpreter *Interpreter) evaluateDefaultDestroyEvent(
+	containingResourceComposite *CompositeValue,
+	eventDecl *ast.CompositeDeclaration,
+) (arguments []Value) {
+	parameters := eventDecl.DeclarationMembers().Initializers()[0].FunctionDeclaration.ParameterList.Parameters
+
+	defer func() {
+		// Only unwind the call stack if there was no error
+		if r := recover(); r != nil {
+			panic(r)
+		}
+		interpreter.SharedState.callStack.Pop()
+	}()
```

Comment:
I guess pushing the invocation onto the call-stack is mostly useful for nicer error reporting?

Eventually we should have equivalents for `invokeInterpretedFunction` and `invokeInterpretedFunctionActivated` for host functions. Currently, invocations of host functions don't push the invocation onto the call-stack, so code like this (and the push at the call-site of `evaluateDefaultDestroyEvent`) are needed